src/Control/Middleware/HTTPMiddlewareAware.php

```diff
@@ -45,11 +48,11 @@ public function addMiddleware(HTTPMiddleware $middleware)
     /**
      * Call middleware
      *
-     * @param $request The request to pass to the middlewares and callback
-     * @param $last The callback to call after all middlewares
+     * @param HTTPRequest $request The request to pass to the middlewares and callback
+     * @param callable $last The callback to call after all middlewares
      * @return HTTPResponse
      */
-    public function callMiddleware(HTTPRequest $request, callable $last)
+    protected function callMiddleware(HTTPRequest $request, callable $last)
```

Comment:
Reasoning is that it's not the responsibility of external classes to decide whether to apply middleware or not; public api of clasess which `use HTTPMiddlewareAware` should internally enforce this, instead of providing multiple public options. :)